saltproc/saltproc.py

```diff
@@ -55,103 +71,280 @@ def __init__(self, steps, cores, nodes, bw, exec_path, restart=False,
+                rep_scheme[group]['from'] = 'fertile'
+            if 'to' not in spec.keys():
+                rep_scheme[group]['to'] = 'waste'
+            if 'eff' not in spec.keys():
+                rep_scheme[group]['eff'] = 1
+
+            # normalize composition
+            if 'comp' in spec.keys():
+                rep_scheme[group]['comp'] = [
+                    x / sum(rep_scheme[group]['comp']) for x in rep_scheme[group]['comp']]
+
+            # check for input errors
+            if 'element' not in spec.keys():
+                raise ValueError('Missing elements for %s' % group)
+            if 'from' not in spec.keys() and 'to' not in spec.keys():
+                raise ValueError('Missing to AND from for %s' % group)
+            if spec['from'] == 'fertile' and 'comp' not in spec.keys():
+                raise ValueError('Must define composition for input material')
+            if spec['to'] == 'waste':
+                for el in spec['element']:
+                    if any(char.isdigit() for char in el):
+                        raise ValueError('You can only remove Elements')
+        self.rep_scheme = rep_scheme
 
     def find_iso_indx(self, keyword):
-        """ Returns index number of keword in bumat dictionary
+        """ Returns index number of keyword in bumat dictionary
 
         Parameters:
         -----------
         keyword: string or list
-            keyword to search for - element (e.g. Xe) or isotope (e.g. Pa233)
+            list for searching element
+            string for isotope
 
         Returns:
         --------
         numpy array of indices
         """
-        indx = 0
         indx_list = []
+        indx = 0
         if isinstance(keyword, str):
-            for key in self.bumat_dict.keys():
-                if keyword in key:
-                    indx_list.append(indx)
-                indx += 1
+            indx = self.isoname.index(keyword)
+            indx_list.append(indx)
         elif isinstance(keyword, list):
-            for key in self.bumat_dict.keys():
-                for keywor in keyword:
-                    if keywor in key:
+            for key in keyword:
+                for indx, isotope in enumerate(self.isoname):
+                    el = " ".join(re.findall("[a-zA-Z]+", isotope))
+                    if key == el:
                         indx_list.append(indx)
-                indx += 1
-
         return np.array(indx_list)
 
+    def get_library_isotopes(self):
+        """ Returns the isotopes in the cross section library
+
+        Parameters:
+        -----------
+
+        Returns:
+        --------
+        iso_array: array
+            array of isotopes in cross section library:
+        """
+        # check if environment variable is set
+        if os.environ.get('SERPENT_DATA') is not None:
+            path = os.environ['SERPENT_DATA']
+        else:
+            path = ''
+        with open(self.input_file, 'r') as f:
```

Comment:
Getting the acelib name should be its own function.
Also, parsing an input file line by line every time you need one of the input parameters is bad practice. Can argparse and objects be used instead?